README.md

```diff
@@ -2,60 +2,70 @@
   <a href="https://github.com/actions/checkout"><img alt="GitHub Actions status" src="https://github.com/actions/checkout/workflows/test-local/badge.svg"></a>
 </p>
 
-# Checkout
+# Checkout V2 (preview)
 
-This action checks out your repository to `$GITHUB_WORKSPACE`, so that your workflow can access the contents of your repository.
+This action checks-out your repository under `$GITHUB_WORKSPACE`, so your workflow can access it.
 
-By default, this is equivalent to running `git fetch` and `git checkout $GITHUB_SHA`, so that you'll always have your repo contents at the version that triggered the workflow.
-See [here](https://help.github.com/en/articles/events-that-trigger-workflows) to learn what `$GITHUB_SHA` is for different kinds of events.
+By default, the repository that triggered the workflow is checked-out, for the ref/SHA that triggered the event.
 
-# Usage
+Refer [here](https://help.github.com/en/articles/events-that-trigger-workflows) to learn which commit `$GITHUB_SHA` points to for different events.
 
-See [action.yml](action.yml)
+Refer [here](https://github.com/actions/checkout/blob/v1/README.md) for previous versions.
 
-Basic:
+# Usage
 
+<!-- start usage -->
```

Comment:
i added a script to regenerate this from the action.yml

the CI verifies the docs are up to date